modules/CmpdReg/src/client/custom/Lot_Custom.js

```diff
@@ -160,19 +163,76 @@ $(function() {
       if (attr.project != null && typeof(attr.project) == 'undefined'){
         errors.push({'attribute': 'project', 'message':  "Project must be provided"});
       }
+      if(this.isNew() & !this.get('isVirtual')) {
+        if("lotNumber" in attr) {
+          if(window.configuration.metaLot.requireLotNumber && attr.lotNumber == null) {
+             errors.push({'attribute': 'lotNumber', 'message':  "Please fill in Lot Number"});
+          }
+          if (attr.lotNumber!=null) {
+            if(isNaN(attr.lotNumber) && attr.lotNumber!='') {
+              errors.push({'attribute': 'lotNumber', 'message':  "Lot Number must be an integer"});
+            } else {
+              if(attr.lotNumber == 0) {
+                errors.push({'attribute': 'lotNumber', 'message':  "Lot Number 0 is reserved for virtual lots"});
+              } else {
+                lotNumbers = this.get('lotNumbers');
+                if(lotNumbers.includes(attr.lotNumber)) {
+                  errors.push({'attribute': 'lotNumber', 'message':  "This lot number is already taken by one of the lots for this compound "+lotNumbers.join(',')});
+                }
+                //  else {
```

Comment:
Can this section of commented-out code be removed?

Reply:
I think you are referring to the commented out part. I actually should just add some comments around it.  It's an example of how one might allow the manual assignment of lot numbers greater than some max number configuration, but anything less than this number must be auto assigned.